docs/source/code-generation/codegen-configuration.mdx

```diff
@@ -0,0 +1,359 @@
+		"name": "TestSubscription",
+		"source": "subscription TestSubscription {\\n  watched\\n}"
+	}
+}
+```
+
+</ExpansionPanel>
+
+### Output options
+
+The code generation engine supports a number of configuration options to change the behaviour of the generator and tailor the generated Swift code to your specific needs.
+
+The top-level properties are:
+
+| Property Name | Description |
+| ------------- | ----------- |
+| `additionalInflectionRules` | Any non-default rules for pluralization or singularization of type names. |
+| `queryStringLiteralFormat` | Formatting of the GraphQL query string literal that is included in each generated operation object. |
+| `deprecatedEnumCases` | Annotate generated Swift enums with the Swift `@available` attribute for GraphQL enum cases annotated with the built-in [`@deprecated` directive](https://spec.graphql.org/draft/#sec--deprecated). |
+| `schemaDocumentation` | Include or exclude [schema documentation](https://spec.graphql.org/draft/#sec-Descriptions) in the generated files. |
+| `apqs` | Whether the generated operations should use Automatic Persisted Queries. |
+| `cocoapodsCompatibleImportStatements` | Generate import statements that are compatible with including `Apollo` via Cocoapods. |
+| `warningsOnDeprecatedUsage` | Annotate generated Swift code with the Swift `@available` attribute and `@deprecated` argument for parts of the GraphQL schema annotated with the built-in [`@deprecated` directive](https://spec.graphql.org/draft/#sec--deprecated). |
+| `conversionStrategies` | Rules for how to convert the names of values from the schema in generated code. |
+| `pruneGeneratedFiles` | Whether unused generated files will be automatically deleted. |
+
+<ExpansionPanel title="Sample JSON">
+
+```json
+"options": {
+	"additionalInflectionRules": [{
+		"pluralization": {
+			"replacementRegex": "animals",
+			"singularRegex": "animal"
+		}
+	}],
+	"queryStringLiteralFormat": "multiline",
+	"deprecatedEnumCases": "include",
+	"schemaDocumentation": "include",
+	"apqs": "disabled",
+	"cocoapodsCompatibleImportStatements": false,
+	"warningsOnDeprecatedUsage": "include",
+	"conversionStrategies": {
+		"enumCases": "camelCase"
+	},
+	"pruneGeneratedFiles": true
+}
+```
+
+</ExpansionPanel>
+
+<ExpansionPanel title="Sample Swift">
+
+```swift
+let configuration = ApolloCodegenConfiguration(
+	// Other properties not shown
+	options: ApolloCodegenConfiguration.OutputOptions(
+		additionalInflectionRules: [.pluralization(
+			singularRegex: "animal",
+			replacementRegex: "animals")],
+		queryStringLiteralFormat: .multiline,
+		deprecatedEnumCases: .include,
+		schemaDocumentation: .include,
+		apqs: .disabled,
+		cocoapodsCompatibleImportStatements: false,
+		warningsOnDeprecatedUsage: .include,
+		conversionStrategies: ApolloCodegenConfiguration.ConversionStrategies(
+			enumCases: .camelCase),
+		pruneGeneratedFiles: true)
+)
+```
+
+</ExpansionPanel>
+
```

Comment:
This is probably a good place for code samples showing how these output options change the generated code. It may get quite long so we'd either need to wrap then in expansion panels or take it out to a separate doc.

I feel this is also something we can do post Summit.